### ISSUE TITLE
Changes to XAIR_Command.c so it can compile in TERMUX on Android

### ISSUE THREAD

**Issue by Rippert** [pmaillot/X32-Behringer#9]
Patrick, I know I should learn how to make a proper pull request, but I found an issue compiling XAIR_Command.c in a TERMUX session on Android. It does compile, but has a bug where the input buffer constantly overflows with "**Line too long...**" errors.

It turns out that the gcc version in TERMUX is very stingy and only gives an **unsigned char** when a variable is declared as **char**. this made _input_ch_ see the -1 value in the non-blocking stdin read come out as a +1, so the buffer filled with fake values. 

I tried changing _input_ch_ to a **signed char**, but those only go to +127, not +255, so that didn't work with your command switch input section.

I ended up making _input_ch_ an **int**, and using it in the stdin reading section. I also made a new variable, _input_ch1_ as a **char**, which I replaced in the command switch section.

Now everything compiles and runs great, on an Android tablet no less. This should all also work on OS X or LINUX, but I'm not sure about Windows, and I don't have a Windows machine. So I'm not sure that this is a good change overall, but if it works in Windows, then you might consider it.

-Ted

**Comment by pmaillot** [comment]
Ted,

Thanks. I made changes in both XAir_Command and X32_Command to reflect your notes. I tested with X32_Command under Windows. Let me know if you find issues with XAir_Command in other architectures. I'll compile under MacOS sometimes today.

-Patrick

**Comment by pmaillot** [comment]
Changes made for 2 weeks. no negative feedback. closing the issue

**Comment by Rippert** [comment]
Patrick, sorry for the delay, the Cold and Flu fairies have been visiting. 

You need to make one more change to your code in both XAir_Command.c and X32_Command.c (I tried it with your X32 emulator).

The issue is with line 502 in XAir_Command.c:
`if (input_ch != NO_CHAR)` 

line 502 in XAir_Command.c and line 382 in X32_Command.c

So you need to change all instances of **input_ch** to **input_intch** from line 468 and above (keyboard input loop) in XAir_Command.c (line 348 and above in X32_command.c).

If you don't mind, I will fork your repository and start working on modifications and bug fixes (if any) on my own, and just try to make pull requests when they seem appropriate. I'm a noob with git, but I'll try to not be too annoying.

**Comment by pmaillot** [comment]
Hello Ted,

 

No problem; I made changes and hope this time they are OK. Funny thing is I didn’t see erroros on my side ‘cause the gcc compiler on PC and Mac are just happy with mismatch between char, unsigned char and int.

 

By all means do not hesitate to fork, modify and propose ideas/changes that will help functionality or bring better support 😊.

 

I’ll be glad to pull in changes.

 

-Patrick-Gilles Maillot

**Comment by pmaillot** [comment]
Ted,

 

I had to undo the changes I brought this morning as they ‘killed’ the immediate mode input under Windows.

 

I suggest you do fork the code and find the right compiler mode or way to get the program(s) to work in your android environment. We can add #ifdefs 😊

 

-Patrick-Gilles Maillot

 

From: Patrick-Gilles Maillot [mailto:patrick.maillot@gmail.com] 
Sent: Saturday, January 13, 2018 10:13 AM
To: 'pmaillot/X32-Behringer' <reply@reply.github.com>; 'pmaillot/X32-Behringer' <X32-Behringer@noreply.github.com>
Cc: 'State change' <state_change@noreply.github.com>
Subject: RE: [pmaillot/X32-Behringer] Changes to XAIR_Command.c so it can compile in TERMUX on Android (#9)

 

Hello Ted,

 

No problem; I made changes and hope this time they are OK. Funny thing is I didn’t see erroros on my side ‘cause the gcc compiler on PC and Mac are just happy with mismatch between char, unsigned char and int.

 

By all means do not hesitate to fork, modify and propose ideas/changes that will help functionality or bring better support 😊.

 

I’ll be glad to pull in changes.

 

-Patrick-Gilles Maillot

**Comment by pmaillot** [comment]
OK, I think I fixed it… v 1.37 (XAir) and 1.35 (X32) are on line .

 

-Patrick-Gilles Maillot

 

From: Patrick-Gilles Maillot [mailto:patrick.maillot@gmail.com] 
Sent: Saturday, January 13, 2018 1:39 PM
To: 'pmaillot/X32-Behringer' <reply@reply.github.com>; 'pmaillot/X32-Behringer' <X32-Behringer@noreply.github.com>
Cc: 'State change' <state_change@noreply.github.com>
Subject: RE: [pmaillot/X32-Behringer] Changes to XAIR_Command.c so it can compile in TERMUX on Android (#9)

 

Ted,

 

I had to undo the changes I brought this morning as they ‘killed’ the immediate mode input under Windows.

 

I suggest you do fork the code and find the right compiler mode or way to get the program(s) to work in your android environment. We can add #ifdefs 😊

 

-Patrick-Gilles Maillot

 

From: Patrick-Gilles Maillot [mailto:patrick.maillot@gmail.com] 
Sent: Saturday, January 13, 2018 10:13 AM
To: 'pmaillot/X32-Behringer' <reply@reply.github.com>; 'pmaillot/X32-Behringer' <X32-Behringer@noreply.github.com <mailto:X32-Behringer@noreply.github.com> >
Cc: 'State change' <state_change@noreply.github.com <mailto:state_change@noreply.github.com> >
Subject: RE: [pmaillot/X32-Behringer] Changes to XAIR_Command.c so it can compile in TERMUX on Android (#9)

 

Hello Ted,

 

No problem; I made changes and hope this time they are OK. Funny thing is I didn’t see erroros on my side ‘cause the gcc compiler on PC and Mac are just happy with mismatch between char, unsigned char and int.

 

By all means do not hesitate to fork, modify and propose ideas/changes that will help functionality or bring better support 😊.

 

I’ll be glad to pull in changes.

 

-Patrick-Gilles Maillot
